### ISSUE TITLE
Where did you set is_train=False for batch_normalization op?

### ISSUE THREAD

**Issue by YunseokJANG** [whxbergkamp/RobustDL_GAN#1]
I tried to run your code with the command for  "run_gan_cifar10.py" on README.md (without changing any parameters) but failed to reproduce the performance in CIFAR-10 dataset (expected: 44.28% but mine shows around 33%). So, as a sanity check, I tried to train the code without the evaluation ops and evaluated over the final checkpoint. However, it gives even less performance than I expected. Thus, I look into the code but failed to find the place of providing "is_train=False" 

https://github.com/whxbergkamp/RobustDL_GAN/blob/a5e1b69c17b2b371c6645c880aa85c776c501947/cifar10/adversarial_networks/models/resnet18_2.py#L248 

for the batch_normalization operation during the evaluation phase 

https://github.com/whxbergkamp/RobustDL_GAN/blob/a5e1b69c17b2b371c6645c880aa85c776c501947/cifar10/adversarial_networks/train_gan_cifar10.py#L5-L109

https://github.com/whxbergkamp/RobustDL_GAN/blob/a5e1b69c17b2b371c6645c880aa85c776c501947/cifar10/adversarial_networks/train_gan_cifar10.py#L288-L292

 Could you take a look and share where did you feed is_train=False? It would be even better if you can share the checkpoint for the trained classifier.

Regards,
Yunseok

-------

[Update] The author mentioned in [Acknowledgement](https://github.com/whxbergkamp/RobustDL_GAN/blob/master/README.md#acknowledgments) that this code is based on [another repo](https://github.com/LMescheder/TheNumericsOfGANs), so I take a look into that repository also. In the original repository, they use another op to measure the performance of a generator, with 'is_training=False'. I put some links below for a better comparison.

https://github.com/LMescheder/TheNumericsOfGANs/blob/68d915fc01608e7f585af853a2aabeacbfa2d53f/consensus_gan/train.py#L26

https://github.com/LMescheder/TheNumericsOfGANs/blob/68d915fc01608e7f585af853a2aabeacbfa2d53f/consensus_gan/train.py#L108-L110

**Comment by chunnamyu** [comment]
Could you share what version of Tensorflow you're running this on and your CUDA version? We will look into this.

**Comment by YunseokJANG** [comment]
Sure, I've tested with the following setting:
Tensorflow 1.13.1 CUDA 10.0 CNDNN 7.6 

However, I'd like to note that feeding   'is_training=True/False'  is orthogonal to Tensorflow versioning issue, as far as I understand. As a sanity check, please run the code with commenting out all the "test" operations. I marked them below:
  https://github.com/whxbergkamp/RobustDL_GAN/blob/a5e1b69c17b2b371c6645c880aa85c776c501947/cifar10/adversarial_networks/train_gan_cifar10.py#L288-L292 

https://github.com/whxbergkamp/RobustDL_GAN/blob/a5e1b69c17b2b371c6645c880aa85c776c501947/cifar10/adversarial_networks/train_gan_cifar10.py#L330-L339

**Comment by chunnamyu** [comment]
What is the weight_decay parameter used in your training? I just noticed the example command in the README uses 1E-4, which is not optimal. The default setting of 1E-5 in the file run_gan_cifar10.py should reproduce the results in the paper. I have also updated the README accordingly.

**Comment by YunseokJANG** [comment]
@chunnamyu  

Yes, I've tested with weight decay 1E-4, so I tested with 1E-5 also now but still failed to replicate.

But, the parameter setting is not seriously related to what I am asking. What I am asking is... "Because of no 'is_train=False' in the training code, the evaluation ops updates running variables (mean, var) from the test dataset, which is incorrect". So, please train your model WITHOUT your evaluation script (L330-339 of train_gan_cifar10.py), and evaluate your model SEPARATELY from your current 'train_gan_cifar10.py'.

**Comment by YunseokJANG** [comment]
Hi, I tested with the evaluation script, but with providing 'is_training=False' to the generator/discriminator in test scripts: https://github.com/whxbergkamp/RobustDL_GAN/blob/a5e1b69c17b2b371c6645c880aa85c776c501947/cifar10/adversarial_networks/train_gan_cifar10.py#L5-L109 .

Unfortunately. this case failed to replicate the performance on CIFAR10, once again. Please recheck your implementation and update the code accordingly.

**Comment by chunnamyu** [comment]
Sorry for the late reply. Thanks for the information on your tests. We are looking into the issue, it might have to do with the decay parameter in batch norm.

**Comment by dssrgu** [comment]
Hello, any updates on this issue? Is the code wrong (including the results in the paper)?
